core/src/main/java/io/questdb/griffin/engine/functions/groupby/CountDistinctLong256GroupByFunction.java

```diff
@@ -32,59 +32,63 @@
 import io.questdb.griffin.engine.functions.GroupByFunction;
 import io.questdb.griffin.engine.functions.LongFunction;
 import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.griffin.engine.groupby.GroupByAllocator;
+import io.questdb.griffin.engine.groupby.GroupByLong256HashSet;
 import io.questdb.std.*;
 
 public class CountDistinctLong256GroupByFunction extends LongFunction implements UnaryFunction, GroupByFunction {
     private final Function arg;
-    private final int setInitialCapacity;
-    private final double setLoadFactor;
-    private final ObjList<Long256HashSet> sets = new ObjList<>();
-    private int setIndex;
+    private final GroupByLong256HashSet setA;
+    private final GroupByLong256HashSet setB;
     private int valueIndex;
 
     public CountDistinctLong256GroupByFunction(Function arg, int setInitialCapacity, double setLoadFactor) {
         this.arg = arg;
-        this.setInitialCapacity = setInitialCapacity;
-        this.setLoadFactor = setLoadFactor;
+        // We use zero as the default value to speed up zeroing on rehash.
+        setA = new GroupByLong256HashSet(setInitialCapacity, setLoadFactor, 0);
+        setB = new GroupByLong256HashSet(setInitialCapacity, setLoadFactor, 0);
     }
 
     @Override
     public void clear() {
-        sets.clear();
-        setIndex = 0;
+        setA.resetPtr();
+        setB.resetPtr();
     }
 
     @Override
     public void computeFirst(MapValue mapValue, Record record) {
-        final Long256HashSet set;
-        if (sets.size() <= setIndex) {
-            sets.extendAndSet(setIndex, set = new Long256HashSet(setInitialCapacity, setLoadFactor));
-        } else {
-            set = sets.getQuick(setIndex);
-            set.clear();
-        }
+        final Long256 l256 = arg.getLong256A(record);
 
-        Long256 val = arg.getLong256A(record);
-        if (isNotNull(val)) {
-            set.add(val.getLong0(), val.getLong1(), val.getLong2(), val.getLong3());
+        if (isNotNull(l256)) {
             mapValue.putLong(valueIndex, 1L);
+            long l0 = l256.getLong0() == 0 ? Numbers.LONG_NaN : l256.getLong0();
```

Comment:
nit: would be great to have `// Remap zero since it's used as the no entry key.` comment here and in `computeNext`.